Quicksearch: use class icons when available

NiklasBr opened the pull request:
## Changes in this pull request  
Show custom class icon in quicksearch for root products (variants still keep their icon)

## Additional info  

![image](https://user-images.githubusercontent.com/279826/122597133-3c7f8400-d06b-11eb-9fa2-3eeb9b0fb0f7.png)

dvesh3 committed:
[Admin] Add Ace editor javascript library for text files preview

dvesh3 committed:
[Admin] Add Ace editor javascript library for text files preview - modification docs

dvesh3 committed:
[Admin] Add Ace editor javascript library for text files preview - destroy with parent

dvesh3 committed:
Merge branch '10.x' into ace_editor

dvesh3 committed:
[Docs] Pimcore X - Redirects

dvesh3 committed:
[Admin] User - Fix roles & docTypes not selected  - resolves #9476 (#9484)

dvesh3 committed:
[Documents] Blocks - fix data loss when using similar names for block and editables - resolves #9421

dvesh3 committed:
[Admin] Fix IndexActionSettingsEvent addSetting not showing up in pimcore.settings - resolves #9396

dvesh3 committed:
[Assets] Video - do not generate versions when processing video thumbnails - resolves #9379 (#9494)

Commit:
correct oder of latitude and longitude in version preview (#9495)

NiklasBr committed:
Class icon in SearchController

NiklasBr committed:
Class icon in SearchController

NiklasBr committed:
Show custom class icon in startup.js

NiklasBr committed:
Show custom class icon in startup.js

NiklasBr committed:
ExtJs tpl syntax

NiklasBr committed:
ExtJs tpl syntax

matthiashamacher committed:
Add Notice for getThumbnail-Function

NiklasBr commented:
These errors make no sense considering the changes:
```log
Error: ] Found 21 errors                                                        

Error: Instantiated class Google_Service_Customsearch not found.
Error: Access to property $cse on an unknown class Google_Service_Customsearch.
Error: Parameter $googleResponse of method Pimcore\Google\Cse::readGoogleResponse() has invalid typehint type Google_Service_Customsearch_Search.
Error: Parameter $googleResponse of method Pimcore\Google\Cse::readGoogleResponse() has invalid typehint type Google_Service_Customsearch_Search.
Error: Call to method getSearchInformation() on an unknown class Google_Service_Customsearch_Search.
Error: Call to method getItems() on an unknown class Google_Service_Customsearch_Search.
Error: PHPDoc tag @var for variable $results contains unknown class Google_Service_Customsearch_Result.
Error: Call to method getPagemap() on an unknown class Google_Service_Customsearch_Result.
Error: Access to offset 'pagemap' on an unknown class Google_Service_Customsearch_Result.
Error: Access to offset 'pagemap' on an unknown class Google_Service_Customsearch_Result.
Error: Access to offset 'pagemap' on an unknown class Google_Service_Customsearch_Result.
Error: Access to offset 'pagemap' on an unknown class Google_Service_Customsearch_Result.
Error: Access to offset 'pagemap' on an unknown class Google_Service_Customsearch_Result.
Error: Call to method getLink() on an unknown class Google_Service_Customsearch_Result.
Error: Property Pimcore\Google\Cse\Item::$raw has unknown class Google_Service_Customsearch_Result as its type.
Error: Parameter $data of method Pimcore\Google\Cse\Item::__construct() has invalid typehint type Google_Service_Customsearch_Result.
Error: Parameter $data of method Pimcore\Google\Cse\Item::__construct() has invalid typehint type Google_Service_Customsearch_Result.
Error: Parameter $data of method Pimcore\Google\Cse\Item::setValues() has invalid typehint type Google_Service_Customsearch_Result.
Error: Parameter $data of method Pimcore\Google\Cse\Item::setValues() has invalid typehint type Google_Service_Customsearch_Result.
Error: Parameter $raw of method Pimcore\Google\Cse\Item::setRaw() has invalid typehint type Google_Service_Customsearch_Result.
Error: Return typehint of method Pimcore\Google\Cse\Item::getRaw() has invalid type Google_Service_Customsearch_Result.
Error: Process completed with exit code 1.
```

dvesh3 commented:
@NiklasBr seems like they changed the namespace in `googleapis/google-api-php-client-services` on v0.2 so stan is unable to find the class. we are on it!

dvesh3 committed:
Add to conflicts "laminas/laminas-code": "<4.4" to solve Symfony 3.4 class compatibility issue

dvesh3 committed:
follow up to #9511

dvesh3 committed:
Add to conflicts "google/apiclient-services": ">=0.200.0" to solve Google Custom Search API class not found error

dvesh3 committed:
Merge remote-tracking branch 'origin/6.9' into 10.0

# Conflicts:
#	composer.json

dvesh3 committed:
Merge remote-tracking branch 'origin/10.0' into 10.x

# Conflicts:
#	README.md

dvesh3 committed:
Merge branch '10.x' of https://github.com/pimcore/pimcore into 10.x

blankse committed:
Fix: Avoid error logs with PPTX files and LibreOffice/Ghostscript

NiklasBr committed:
Remove problematic URLSlug validation (#9171)

* Remove problematic URLSlug validation

* Remove invalid test case

* New changes to URL Slug validation

* Fixes lost negation in if

* Do not use polyfilled PHP8 string functions for Symfony 3.4 support

brusch committed:
Merge remote-tracking branch 'origin/6.9' into 10.0

brusch committed:
Merge remote-tracking branch 'origin/10.0' into 10.x

ascheider committed:
Fix custom views loading. (#9491)

* Fix custom views loading.
Backport of: https://github.com/pimcore/pimcore/issues/8900

* Update models/DataObject/Listing/Dao.php

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>
Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

brusch committed:
Merge remote-tracking branch 'origin/6.9' into 10.0

NiklasBr committed:
Fix inherited relation values not being visible after path formatting

brusch committed:
Merge remote-tracking branch 'origin/6.9' into 10.0

# Conflicts:
#	lib/Document/Adapter/LibreOffice.php

brusch committed:
Merge remote-tracking branch 'origin/6.9' into 10.0

brusch committed:
[Permissions] Properly escape type variable - fixes #9412

mugge6 committed:
Add fix for PHP8 compatibility (#9527)

blankse committed:
Fix wrong Password hint

brusch committed:
Merge remote-tracking branch 'origin/6.9' into 10.0

brusch committed:
Kernel getContainer() throws a LogicException on shutdown - fixes #9531

dvesh3 committed:
[ControllerDataProvider] Fix bundle templates in getTemplates() calls (#9454)

* [ControllerDataProvider] Fix bundle templates in getTemplates() calls - resolves #9453

* Update lib/Controller/Config/ControllerDataProvider.php

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

brusch commented on bundles/AdminBundle/Controller/Searchadmin/SearchController.php:
```diff
@@ -482,6 +482,14 @@ public function quicksearchAction(Request $request, EventDispatcherInterface $ev
                     $data['iconCls'] .= ' pimcore_icon_' . \Pimcore\File::getFileExtension($element->getFilename());
+                 } elseif ($element instanceof DataObject\Concrete) {
+                    // Show custom on base class (not variants) if it exists
+                    if ($element->getType() === DataObject::OBJECT_TYPE_OBJECT && $element->getClass()->getIcon()) {
```
I think we should use the same concept here as we do for the tree, see: https://github.com/pimcore/pimcore/blob/2f18a0f26380c56b609d937b40d2050c8d1a3e1a/bundles/AdminBundle/Controller/Traits/AdminStyleTrait.php#L37

Commit:
Enrich Layout for CS Attributes in Grid (#9486)

Without enriching the layout definition, batch editing does not work for selects / multiselect with static options. By enriching the layout the options are correctly provided and batch editing also works for select / multiselect.

brusch committed:
Merge remote-tracking branch 'origin/6.9' into 10.0

brusch committed:
Merge remote-tracking branch 'origin/10.0' into 10.x

Commit:
[Application logger] Make text selectable / Link related object (#9512)

* make text selectable in applicaiton logger

* link related object

weisswurstkanone committed:
[DATAOBJECT] Preview tab not working with non empty image #9521

dvesh3 committed:
Merge remote-tracking branch 'origin/10.0' into 10.x

fashxp committed:
fixed #9084 - Update screenshots on Readme.md (#9548)

ubaranouski committed:
fix(kernel) allow tests to finish with cleared container

ubaranouski committed:
fix(kernel) allow tests to finish with cleared container

brusch committed:
Permissions: fix quoting - regression of #9526 - fixes #9555

brusch committed:
Merge remote-tracking branch 'origin/10.0' into 10.x

blankse committed:
Create PHP files with better Code Style (#9474)

* Create PHP definition files with better Code Style

* Remove unused field definition variables, emtpy line before return

* Function opening braces in new line

* Class opening braces in new line

putzflorian committed:
Additional Tools Links has changed

brusch committed:
Merge remote-tracking branch 'origin/10.0' into 10.x

Commit:
Fix video asset image thumbnail parameter

Use setimage instead of settime for image_thumbnail_asset

dturopoli committed:
Localized error pages (#9270)

* Localized error pages

* Apply php-cs-fixer changes

* Localized error pages adjustments

* Sites localized error documents

* Apply php-cs-fixer changes

* If fallback document is the root-document, use browser locale detection

* Apply php-cs-fixer changes

Co-authored-by: dturopoli <dturopoli@users.noreply.github.com>
Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

brusch committed:
[Data Object] Fixed quoting in ClassificationstoreController

brusch committed:
Readme: Removed 6.x from supported list

brusch committed:
Merge remote-tracking branch 'origin/10.0' into 10.x

# Conflicts:
#	README.md

brusch committed:
Revert #8818 - thew problem with this fix is that the preview image in <src> is not displayed anymore with this fix

brusch committed:
[Thumbnail] allow  custom <img loading=""> attribute - fixes #9577

brusch committed:
Clear cache from Admin UI throws exception (#9578)

* Clear cache from Admin UI throws exception - fixes #9575

* fixed PHPDoc

brusch committed:
Apply php-cs-fixer changes

dvesh3 committed:
[Admin] Drafts - hide notification when save permission is missing - resolves #9559

dvesh3 committed:
[Docs] Updates for Pimcore X

mugge6 committed:
Fix issue with missing indexName in ecommerce indexer configuration. (#9545)

cipribucur committed:
Merge pull request #9371 from pimcore/ace_editor

[Admin] Add Ace editor javascript library for text files preview

dvesh3 committed:
[Admin] Server File explorer - use Ace editor (#9382)

* [Admin] Server File explorer - use Ace editor

* [Admin] Robotstxt - use Ace editor

Gerhard-Kanzler committed:
Add possibility to load unpublished objects (#9380)

Add possibility to load unpublished objects with param convter options
/**
* @Route("/blog/{post}")
* @ParamConverter( "post", options={ "unpublished" = true } )
*/

Commit:
followup docs for the #9380 issue - unpublished objects

brusch committed:
Merge remote-tracking branch 'origin/10.0' into 10.x

dvesh3 committed:
[Ghostscript] Fix PDF thumbnail of a page - resolves #9593

fashxp committed:
fixed missing adapter config in pdfreactor

dvesh3 committed:
Fix phpstan https://github.com/pimcore/pimcore/runs/2939469676

dvesh3 committed:
Merge branch '10.0' into 10.x

dvesh3 committed:
[Docs] Pimcore X - Update Google Custom Search Engine docs

dvesh3 committed:
[Core] Fix Pimcore Password encoder not working with Symfony ^5.3 - resolves #9489

BabovicT committed:
[XLIFF] The export removes zeros (#9586) (#9588)

* [XLIFF] The export removes zeros (#9586)

Export correctly exporting zeros now.

* Update lib/Translation/Escaper/Xliff12Escaper.php

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

brusch committed:
[Admin] Minimized scripts should be saved in Flysystem storage (#9606)

* [Admin] Minimized scripts should be saved in Flysystem storage

* added deprecation note

brusch committed:
Apply php-cs-fixer changes

blankse committed:
Can't override/disable image optimizer executable

dvesh3 committed:
[Admin] Grid view - Fix ClassificationStore sorting/filtering is not working -  resolves #9594

blankse committed:
ImageOptimizer: Remove unused mime type guesser

blazevici committed:
[Data Object] WYSIWYG: char counter and maxChar limit in datatype (#9554)

* feat: wysiwyg char counter and maxChar limit in datatype

* fix: remove /samples bloat folder

HaaseIT committed:
added sample command for generating project specific migration (#9623)

* added config for migrations in namespace App\Migrations

* fixed improper path

NiklasBr committed:
Increase colour contrast on locked elements in tree (#9625)

* Increase contrast base WCAG rules

* Increase contrast base WCAG rules

cipribucur committed:
Google apiclient services (#9589)

* fix issue #9513 [Google API] Use proper namespaces for API references

* fix issue #9513 [Google API] api client change

* update google/apiclient to 2.10 version

* update change in docs - resolves #9513

brusch committed:
Apply php-cs-fixer changes

brusch committed:
Merge remote-tracking branch 'origin/10.0' into 10.x

NiklasBr committed:
Use addAdminStyle trait for event dispatch

NiklasBr committed:
Use addAdminStyle trait for event dispatch

NiklasBr committed:
Merge branch 'patch-7' of https://github.com/NiklasBr/pimcore into patch-7

NiklasBr commented:
Rebase:ing is the bane of my existence.
